go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -181,6 +185,57 @@ func (c *addressManager) Run(stopChan <-chan struct{}, doneWg *sync.WaitGroup) {
 	klog.Info("Node IP manager is running")
 }
 
+// updateNodeAddressAnnotations updates all relevant annotations for the node including
+// k8s.ovn.org/host-addresses, k8s.ovn.org/node-primary-ifaddr, k8s.ovn.org/l3-gateway-config.
+func (c *addressManager) updateNodeAddressAnnotations() {
+	// Get node information
+	node, err := c.watchFactory.GetNode(c.nodeName)
+	if err != nil {
+		klog.Errorf("Unable to get node from informer: %v", err)
+		return
+	}
+
+	// get updated interface IP addresses for the gateway bridge
+	ifAddrs, err := getNetworkInterfaceIPAddresses(c.gatewayBridge.bridgeName)
```

Comment:
dont you need to lock the bridge here while you read from it? Also lock the bridge in the other thread?

Reply:
yes, working on it.

Reply:
done